### PR TITLE
WIP: update api for python module, drop python2

### PR DESCRIPTION
**Pull request by simonpintarelli**
- [x] drop python2
- [x] update to new SIRIUS API for most important parts
- [ ] cleanup / split `py_sirius.cpp` into multiple cpp files 
- [x] @toxa81  ci/cd pipeline for python module

### CODE REVIEW

File: spack/packages/sirius/package.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ class Sirius(CMakePackage, CudaPackage):
     depends_on('pkgconfig', type='build')
 
     # Python module
-    depends_on('python', when='+python', type=('build', 'run'))
+    depends_on('python', when='+python@3.6:', type=('build', 'run'))
```
Comment (toxa81) [inline spack/packages/sirius/package.py:98]: IMO we should drop this file as well and submit changes directly to the spack package of sirius

**Comment by toxa81** [comment]
The vector3d<double> intrface is not working properly.   This works but is awkward: 
https://github.com/electronic-structure/SIRIUS/blob/update/python-module/examples/python/Si_dft.py#L29

This should be the intrface (https://github.com/electronic-structure/SIRIUS/blob/update/python-module/examples/python/Si_dft.py#L30), but it gives the error: 

```
TypeError: add_atom(): incompatible function arguments. The following argument types are supported:
    1. (self: sirius.py_sirius.Unit_cell, arg0: str, arg1: geometry3d::vector3d<double>) -> None
```

**Comment by simonpintarelli** [comment]
> The vector3d intrface is not working properly. This works but is awkward: https://github.com/electronic-structure/SIRIUS/blob/update/python-module/examples/python/Si_dft.py#L29
> 
> This should be the intrface (https://github.com/electronic-structure/SIRIUS/blob/update/python-module/examples/python/Si_dft.py#L30), but it gives the error:
> 
> ```
> TypeError: add_atom(): incompatible function arguments. The following argument types are supported:
>     1. (self: sirius.py_sirius.Unit_cell, arg0: str, arg1: geometry3d::vector3d<double>) -> None
> ```

I'll have a look.

**Comment by simonpintarelli** [comment]
@toxa81 Now both variants `Unit_cell.add_atom(label, [list_like])` and `Unit_cell.ad_atom(label, sirius.vector3d)`. The former also does check the length, it will accept only python list / numpy arrays with 3 entries (pybind11 does that).

**Comment by toxa81** [comment]
This looks good to me now. Unless @simonpintarelli you _really_ want to work on `cleanup / split py_sirius.cpp into multiple cpp files`, it is good to go for meging. The Python intrface is restored to a minimum that can run DFT ground state.

**Comment by simonpintarelli** [comment]
@toxa81 Let's merge it. I'll work on it later.

**Pull request merged**